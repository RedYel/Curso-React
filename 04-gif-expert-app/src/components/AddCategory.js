import React, { useState } from 'react';

export const AddCategory = () => {

    const [inputValue, setinputValue] = useState('Haz tu busqueda aquÃ­');

    const handleInputChange = ( e ) => {
        setinputValue( e.target.value );
    }
    
    const handleSubmit = (e) => {
        e.preventDefault();

        console.log( 'Submit Hecho' );
    }

  return (
    <form onSubmit={ handleSubmit }>
        <input
            type="text"
            value={ inputValue }
            onChange={ handleInputChange }
        />
    </form>
  )
};
