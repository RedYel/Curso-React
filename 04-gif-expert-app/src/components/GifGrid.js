import React, { useState, useEffect} from 'react';
import { GifGridItem } from './GifGridItem';

export const GifGrid = ({ category }) => {

    const [images, setImages] = useState([]);

    useEffect(() => {
      getGifs();
    }, [])
    

    const getGifs = async() => {
        const url = 'https://api.giphy.com/v1/gifs/search?q=Batman&limit=10&api_key=7o4oshT61TLIaGK7U8x8XLXu7djzW2kF'
        
        const resp = await fetch( url ) 
        const { data } = await resp.json();
            //console.log(data);
        const gifs = data.map( img => {
            return {
                id: img.id,
                title: img.title,
                url: img.images?.downsized_medium.url
            }
        })

        console.log(gifs);
        setImages( gifs );
    
    }

    //getGifs();

  return (
    <div>
        <h3>{ category }</h3>
        {
            images.map( img => (
                <GifGridItem 
                    key={ img.id }
                    //img={ img } 
                    { ...img }
                />
            ))
        }
    </div>
  )
};
