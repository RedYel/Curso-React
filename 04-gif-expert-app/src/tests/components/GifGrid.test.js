import React from 'react';
import { shallow } from 'enzyme';
import '@testing-library/jest-dom';
import { GifGrid } from '../../components/GifGrid';
import { useFetchGifs } from '../../hooks/useFetchGifs';
jest.mock('../../hooks/useFetchGifs');

describe('Pruebas en el GifGrid', () => {

    const category = 'batman';

    test('Debe de hacer match con el snapshot', () => {

        useFetchGifs.mockReturnValue({
            data: [],
            loading: true
        });

        const wrapper =shallow( <GifGrid category={category}/> );

        expect( wrapper ).toMatchSnapshot();
    });
    
    test('Debe mostrar items cuando se cargan imagenes useFetchGifs', () => {
        
        const gifs = [{
            id: 'ABC',
            url: 'https://localhost.algo',
            title: 'batman loco'
        },
        {
            id: 'Asdg',
            url: 'https://localhost.algo',
            title: 'batman sad'
        }
        ];

        useFetchGifs.mockReturnValue({
            data: gifs,
            loading: false
        });

        const wrapper =shallow( <GifGrid category={category}/> );

        expect( wrapper ).toMatchSnapshot();
        expect( wrapper.find('p').exists() ).toBe(false);
        expect( wrapper.find('GifGridItem').length ).toBe( gifs.length );
    });
    

});
