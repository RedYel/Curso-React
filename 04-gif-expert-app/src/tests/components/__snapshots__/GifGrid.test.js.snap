// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en el GifGrid Debe de hacer match con el snapshot 1`] = `
<Fragment>
  <h3
    className="animate__animated animate__flash"
  >
    batman
  </h3>
  <p
    className="animate__animated animate__flash"
  >
    Loading
  </p>
  <div
    className="card-grid"
  />
</Fragment>
`;

exports[`Pruebas en el GifGrid Debe mostrar items cuando se cargan imagenes useFetchGifs 1`] = `
<Fragment>
  <h3
    className="animate__animated animate__flash"
  >
    batman
  </h3>
  <div
    className="card-grid"
  >
    <GifGridItem
      id="ABC"
      key="ABC"
      title="batman loco"
      url="https://localhost.algo"
    />
    <GifGridItem
      id="Asdg"
      key="Asdg"
      title="batman sad"
      url="https://localhost.algo"
    />
  </div>
</Fragment>
`;
